server/client_server.go

```diff
@@ -22,10 +24,125 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
+type IssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values,omitempty"`
+}
+
+type FieldInfo struct {
+	Values []interface{} `json:"values,omitempty"`
+}
+
+type FieldValues struct {
+	FieldID string `json:"fieldId,omitempty"`
+}
+
+type FieldId struct {
+	Values []FieldValues `json:"values,omitempty"`
+}
+
+type Version struct {
+	VersionInfo string `json:"version,omitempty"`
+}
+
+const (
+	CreateMetaAPIEndpoint = "rest/api/2/issue/createmeta/"
+	PivotVersion          = "8.4.0"
+)
+
 // GetCreateMeta returns the metadata needed to implement the UI and validation of
 // creating new Jira issues.
 func (client jiraServerClient) GetCreateMeta(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, error) {
-	cimd, resp, err := client.Jira.Issue.GetCreateMetaWithOptions(options)
+	ServerInfoApiEndpoint := "rest/api/2/serverInfo"
+	v := new(Version)
+	req, err := client.Jira.NewRequest(http.MethodGet, ServerInfoApiEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = client.Jira.Do(req, v)
+	if err != nil {
+		return nil, err
+	}
+
+	currentVersion, err := version.NewVersion(v.VersionInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	pivotVersion, err := version.NewVersion(PivotVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	var info *jira.CreateMetaInfo
+	var resp *jira.Response
+	if currentVersion.LessThan(pivotVersion) {
+		cimd, response, error := client.Jira.Issue.GetCreateMetaWithOptions(options)
+		info = cimd
+		resp = response
+		err = error
+	} else {
+		cd, response, err := client.Jira.Project.ListWithOptions(options)
+		meta := new(jira.CreateMetaInfo)
+
+		if err == nil {
```

Comment:
you can write this better rather than checking this condition you can check if err != nil and then do everything else without any if blocks

Reply:
What if I don't have to do anything specifically when `err != nil`. And I think there are 3 other places where `err==nil` condition is mentioned.

Reply:
then just return there is no need to write the entire code in an if condition whether it is for err == nil or not

Reply:
Why was this resolved without fixing?

Reply:
On this I had a discussion with @Kshitij-Katiyar , then we mutually decided that for this case `err==nil` condition seems good.